applications/LinearSolversApplication/custom_solvers/amgcl_raw_solver.cpp

```diff
@@ -0,0 +1,748 @@
+            case 1: {                                                                                       \
+                KRATOS_CONSTRUCT_AMGCL_SOLVER_BUNDLE_WITH_BLOCK_SIZE(BACKEND_TEMPLATE, BACKEND_SCALAR, 1);  \
+                break;                                                                                      \
+            }                                                                                               \
+            case 2: {                                                                                       \
+                KRATOS_CONSTRUCT_AMGCL_SOLVER_BUNDLE_WITH_BLOCK_SIZE(BACKEND_TEMPLATE, BACKEND_SCALAR, 2);  \
+                break;                                                                                      \
+            }                                                                                               \
+            case 3: {                                                                                       \
+                KRATOS_CONSTRUCT_AMGCL_SOLVER_BUNDLE_WITH_BLOCK_SIZE(BACKEND_TEMPLATE, BACKEND_SCALAR, 3);  \
+                break;                                                                                      \
+            }                                                                                               \
+            case 4: {                                                                                       \
+                KRATOS_CONSTRUCT_AMGCL_SOLVER_BUNDLE_WITH_BLOCK_SIZE(BACKEND_TEMPLATE, BACKEND_SCALAR, 4);  \
+                break;                                                                                      \
+            }                                                                                               \
+            case 5: {                                                                                       \
+                KRATOS_CONSTRUCT_AMGCL_SOLVER_BUNDLE_WITH_BLOCK_SIZE(BACKEND_TEMPLATE, BACKEND_SCALAR, 5);  \
+                break;                                                                                      \
+            }                                                                                               \
+            case 6: {                                                                                       \
+                KRATOS_CONSTRUCT_AMGCL_SOLVER_BUNDLE_WITH_BLOCK_SIZE(BACKEND_TEMPLATE, BACKEND_SCALAR, 6);  \
+                break;                                                                                      \
+            }                                                                                               \
+            default: KRATOS_ERROR << "unsupported block size: " << mpImpl->mDoFCount << "\n";               \
+        } // switch mpImpl->mDoFCount
+
+    // Construct the solver
+    switch (mpImpl->mBackendType) {
+        #ifdef AMGCL_GPGPU
+            case AMGCLBackendType::SinglePrecisionGPU: {
```

Comment:
@matekelemen , I can not see where the selection between float and double could go wrong, but I would add some debug prints here in each of the cases. And then, if it behaves here as expected, I would add some temporary output inside [amgcl::make_solver](https://github.com/KratosMultiphysics/Kratos/blob/217a930723dc6099d18eb426db783935d8dea917/external_libraries/amgcl/make_solver.hpp#L102) (may be with `sizeof(value_type)` in the output).